circuit/src/execution-delivery/src/exec_composer/mod.rs

```diff
@@ -849,4 +892,303 @@ mod tests {
             )
         });
     }
+
+    #[test]
+    fn pre_run_bunch_until_break_multiple_contracts_succeeds() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let escrow_account = setup_test_escrow_as_tx_signer(&mut ext);
+
+        let compose_one =
+            make_compose_out_of_raw_wat_code::<Test>(CODE_CALL, vec![], requester.clone(), value);
+
+        let compose_two = make_compose_out_of_raw_wat_code::<Test>(
+            WROONG_CODE_MODULE_DISPATCH_NO_FUNC,
+            vec![],
+            requester.clone(),
+            value,
+        );
```

Comment:
Hmm somehow I think this should break for a contract with wrong wasm code?

Reply:
There is another variable called INVALID_CODE. That is wrong wasm code.
This one compiles fine. As per the name suggests, it would fail when it runs. Will find out more.

Reply:
Sounds good!

Reply:
I went through the substrate repo and searched for this particular contract. It is not there. Most probably they have changed it over time.
At this point, I am not sure what this wat does, except that it compiles.